hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -438,12 +438,12 @@ public void start() {
         proxyManager.init(config, clientContext);
         listenerService.start();
         loadBalancer.init(getCluster(), config);
-        partitionService.start();
         statistics.start();
         clientExtension.afterStart(this);
     }
 
     public void onClusterConnect(Connection ownerConnection) throws Exception {
+        partitionService.listenPartitionTable(ownerConnection);
```

Comment:
what happens if event is received between connection established and this point? is there a way to register the listener earlier?

Reply:
In a distributed environment does it matter to get the partition table changes right before you connected ? Since things will work if you connected X ms later and get the changes right after, the delay between creating the connection and getting the changes should not matter right ?